lib/logstash/outputs/opsgenie.rb

```diff
@@ -6,6 +6,27 @@
 require "net/http"
 require "net/https"
 
+
+HTTP_EXCEPTIONS = [
+  Errno::EINVAL,
+  Net::HTTPBadResponse,
+  Net::HTTPHeaderSyntaxError,
+]
+
+RETRYABLE_HTTP_EXCEPTIONS = [
+  OpenSSL::SSL::SSLError,
+  EOFError,
+  Timeout::Error,
+  Errno::ETIMEDOUT,
+  Errno::ECONNRESET,
+  Errno::ECONNREFUSED,
+  Net::ProtocolError,
+  SocketError,
+  EOFError
+]
+
+HTTP_EXCEPTION_RETRIES = 5
```

Comment:
This is such a tiny thing that I don't think we should be spending a lot of time parametrising everything and writing million tests. with 5 retries we get enough attempts spread over 31s.